includes/event.php

```diff
@@ -52,4 +52,26 @@ public function end(): DateTimeImmutable {
 	public function timezone(): DateTimeZone {
 		return $this->timezone;
 	}
+
+	/**
+	 * Generate text for the end date.
+	 *
+	 * @param string $event_end The end date.
+	 *
+	 * @return string The end date text.
+	 */
+	public static function get_end_date_text( string $event_end ): string {
+		$end_date_time     = new DateTimeImmutable( $event_end );
+		$current_date_time = new DateTimeImmutable( 'now', new DateTimeZone( 'UTC' ) );
+
+		$interval       = $end_date_time->diff( $current_date_time );
+		$hours_left     = ( $interval->d * 24 ) + $interval->h;
+		$hours_in_a_day = 24;
+
+		if ( $hours_left <= $hours_in_a_day ) {
+			return 'ends in ' . $hours_left . ' hours';
+		} else {
+			return 'until ' . $end_date_time->format( 'M j, Y' );
```

Comment:
Let's avoid the string concat and use a `sprintf()` with a `__()` and a placeholder.